_data-prepper/pipelines/configuration/processors/mutate-event.md

```diff
@@ -0,0 +1,302 @@
+
+```json
+{"message": "value"}
+```
+
+When you run the `RenameKey` processor, it parses the message into the following output as the "newMessage":
+
+```json
+{"newMessage": "value"}
+```
+
+> If `newMessage` already exists, its existing value is overwritten with `value`.
+
+
+
+### Special considerations
+
+Renaming operations occur in the order that the key pair entries are listed in within the `pipelines.yaml` file. This means that chaining (where key pairs are renamed in sequence) is implicit with the `RenameKey` processor. See the following `piplines.yaml` file example:
+
+```yaml
+pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - rename_key:
+        entries:
+        - from_key: "message"
+          to_key: "message2"
+        - from_key: "message2"
+          to_key: "message3"
+  sink:
+    - stdout:
+```
+
+Add the following contents to the `logs_json.log` file:
+
+```json
+{"message": "value"}
+```
+
+After the processor runs, the following output appears:
+
+```json
+{"message3": "value"}
+```
+
+## ConvertEntry
+
+The `ConvertEntry` processor converts a value type associated with the specified key in a message to the specified type. It is a casting processor that changes the types of some fields in the event or message. Some of inputted data may need to be converted to different types, such as an integer or a double, or a string to an integer, so that it will pass the events through condition-based processors, or to perform conditional routing.
+
+### Configuration
+
+You can configure the `ConvertEntry` processor with the following options.
+
+Option | Required | Description
+:--- | :--- | :---
+| `key`| Yes | Keys whose value needs to be converted to a different type. | 
+| `type` | No | Target type for key value. Possible values are `integer`, `double`, `string`, and `boolean`. Default value is `integer`. |
+
+### Usage
+
+To get started with type conversion processor using Data Prepper, create the following `pipeline.yaml` file:
+
+```yaml
+type-conv-pipeline:
+  source:
+    file:
+      path: "/full/path/to/logs_json.log"
+      record_type: "event"
+      format: "json"
+  processor:
+    - grok:
+        match:
+          message: ['%{IPORHOST:clientip} \[%{HTTPDATE:timestamp}\] %{NUMBER:response_status}']
+    - convert_entry_type:
+        key: "response_status"
+        type: "integer"
+  sink:
+    - stdout:
+```
+
+Next, create a log file named `logs_json.log` and replace the `path` in the file source of your `pipeline.yaml` with this filepath. For more information, see [Configuring Data Prepper]({{site.url}}{{site.baseurl}}/data-prepper/getting-started/#2-configuring-data-prepper).
+
+```json
+{"message": "10.10.10.19 [19/Feb/2015:15:50:36 -0500] 200"}
+```
+
+When you run the `Grok` processor, the processor parses the message into the following output:
```

Comment:
Which processor are we talking about here? Grok or Convert Entry?

Reply:
After seeing the demo, Grok was correct, but deemed not essential in this instance, so this section was updated accordingly.